subsys/net/lib/openthread/platform/radio.c

```diff
@@ -958,12 +958,32 @@ void otPlatRadioSetMacKey(otInstance *aInstance, uint8_t aKeyIdMode,
 	__ASSERT_NO_MSG(aPrevKey != NULL && aCurrKey != NULL &&
 			aNextKey != NULL);
 
+	struct ieee802154_key keys[] = {
+		{
+			.key_id_mode = aKeyIdMode,
```

Comment:
Given that Key Id Mode values are being shifted for OpenThread usage (https://github.com/openthread/openthread/blob/main/src/core/mac/mac_frame.hpp#L355-L359) we should shift them back before passing to the IEEE 802.15.4 driver.